src/libraries/Directory.Build.targets

```diff
@@ -135,7 +135,7 @@
   <Import Project="$(RepositoryEngineeringDir)testing\runsettings.targets" Condition="'$(EnableRunSettingsSupport)' == 'true'" />
   <Import Project="$(RepositoryEngineeringDir)testing\coverage.targets" Condition="'$(EnableRunSettingsSupport)' == 'true' or '$(EnableCoverageSupport)' == 'true'" />
 
-  <Import Project="$(RepositoryEngineeringDir)illink.targets" Condition="'$(IsSourceProject)' == 'true' or '$(ExplicitlyImportCustomILLinkTargets)' == 'true'" />
+  <Import Project="$(RepositoryEngineeringDir)illink.targets" Condition="('$(IsSourceProject)' == 'true' and '$(TargetFramework)' == '$(NetCoreAppCurrent)') or '$(ExplicitlyImportCustomILLinkTargets)' == 'true'" />
```

Comment:
Looking at all these conditions it might make sense to introduce a deferred property like `ImportCustomILLInkTargets` and condition on that instead. That would allow replacing `ExplicitlyImportCustomILLinkTargets` with it and make this more readable and more extensible.